llama_index/pipeline.py

```diff
@@ -0,0 +1,42 @@
+# from functools import
+import json
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field, asdict
+from typing import Dict, List, Any
+
+
+@dataclass
+class PipelineSchema:
+    """Class representing a component in a pipeline."""
+
+    name: str
```

Comment:
I think `name` should actually come from an Enum with str. So that downstream can consume intelligently similar to `EventType`.

Reply:
Let's KIV this.

Reply:
Added TODO